config/src/main/java/org/springframework/security/config/annotation/web/HttpConfigurerConsumer.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.config.annotation.web;
+
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+
+/**
+ * Callback interface used to process an {@link AbstractHttpConfigurer} with the ability to
+ * throw a (checked) exception.
+ *
+ * @author Eleftheria Stein
+ * @param <C> the configurer type
+ */
+@FunctionalInterface
+public interface HttpConfigurerConsumer<C extends AbstractHttpConfigurer> {
```

Comment:
The `java.util.function.Consumer` does't allow exceptions to be thrown from the `accept` method.
In our case the `realmName` method on `HttpBasicConfigurer` throws an exception, so you would not be able to use that method in a lambda with the Java `Consumer`.
I have seen a custom consumer in Spring Boot and I assume it is for the same reason. 
https://github.com/spring-projects/spring-boot/blob/36c1c051b8106371d37f8a747cea0e369a8e5c84/spring-boot-project/spring-boot-test/src/main/java/org/springframework/boot/test/context/runner/ContextConsumer.java

Reply:
Thanks for the pointer. That makes sense. A few thoughts:

* I think we should change the signature to allow throwing `Throwable` rather than `Exception`
* I'm thinking we may want to make this more central (i.e. within core) since it is likely we will need to use this in other places.

Reply:
@rwinch I moved the consumer to the core module and renamed it. I also made the `Throwable` a type argument so that it doesn't need to bubble up as a `Throwable` in the calling methods.